### PR TITLE
Ast refactor

### PR DESCRIPTION
**Pull request by kenarab**
It has a dependency on PR #139.

It includes renaming structs for having a better convention for understanding code.

### CODE REVIEW

File: crates/concrete_ast/src/statements.rs [hunk 1]
```diff
@@ -18,15 +18,15 @@ pub enum Statement {
...
 pub enum LetStmtTarget {
-    Simple { name: Ident, r#type: TypeSpec },
+    Simple { id: Ident, r#type: TypeSpec },
     Destructure(Vec<Binding>),
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct LetStmt {
     pub is_mutable: bool,
-    pub target: LetStmtTarget,
```
Comment (juanbono) [inline crates/concrete_ast/src/statements.rs:28]: here `target` and `value` sound more understandable.

File: crates/concrete_ast/src/expressions.rs [hunk 2]
```diff
@@ -99,16 +99,16 @@ pub enum BitwiseOp {
 
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct MatchExpr {
-    pub value: Box<Expression>,
+    pub match_expr: Box<Expression>,
```
Comment (juanbono) [inline crates/concrete_ast/src/expressions.rs:102]: why not just `expr`?

File: crates/concrete_ast/src/statements.rs [hunk 3]
```diff
@@ -38,30 +38,30 @@ pub struct ReturnStmt {
...
-    pub name: Ident,
+    pub id: Ident,
     pub rename: Option<Ident>,
     pub r#type: TypeSpec,
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct ForStmt {
     pub init: Option<LetStmt>,
-    pub condition: Option<Expression>,
```
Comment (juanbono) [inline crates/concrete_ast/src/statements.rs:57]: I think `condition` is better than `cond`

**Review by juanbono — CHANGES_REQUESTED** [review]
See comments

**Review by juanbono — APPROVED** [review]

**Comment by kenarab** [comment]
> > See comments
> 
> Applied all suggestions. There is still an open question to a comment

All reviewed changes closed

**Review by igaray — APPROVED** [review]

**Pull request merged**